### ISSUE TITLE
The type of the identifier of an if statement is inferred as `Never` when pyright tries to exhaust a sum type value

### ISSUE THREAD

**Issue by Congee** [microsoft/pyright#1000]
**Describe the bug**
Title says it all.

**To Reproduce**
See code below.

**Expected behavior**
At least the type inference should consistently `Never` _or_ `Literal[2, 3]` not both. Ideally, I would like it to be `Literal[2, 3]` without any error report (I used to see this behavior).

**Screenshots or Code**
```python
class C:
    member: Literal[1, 2, 3] = 1

c = C()
c.member
if c.member == 1:
    ...
elif c.member == 2:  # pyright infers correctly the type of member `Literal[2, 3]`. However, it reports Cannot access member "member" for type "Never", too
    ...
```

```
d: Literal[1,2,3] = 1

if d == 1:
    ...
elif d == 2:  # here, d is inferred as `Never`
    ...
```
![image](https://user-images.githubusercontent.com/2083950/92125454-0ff4c700-edcd-11ea-98a1-58148030151d.png)
![image](https://user-images.githubusercontent.com/2083950/92125471-14b97b00-edcd-11ea-8f6c-69beb6fb5c43.png)

**VS Code extension or command-line**
coc-pyright with pyright 1.1.65

**Additional context**
N/A

**Comment by Congee** [comment]
Issue number 1000. Yay!

**Comment by erictraut** [comment]
Thanks for the bug report. This was caused by a bug in the logic I introduced for "discriminated union type narrowing" (#944). It will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.67, which I just published. It will also be included in next week's release of Pylance.
